test/parallel/test-dgram-udp6-link-local-address.js

```diff
@@ -5,58 +5,49 @@ if (!common.hasIPv6)
 
 const assert = require('assert');
 const dgram = require('dgram');
-
 const os = require('os');
-const ifaces = os.networkInterfaces();
-
-// Try to an interface with an IPv6 link local address
-const address = Object.keys(ifaces)
-
-// Convert the interface names to the interface arrays of addresses
-      .map(
-        (ifname) => ifaces[ifname]
-          .map((ifobject) => Object.assign({ ifname: ifname }, ifobject)))
-
-  // Flatten the array of arrays
-      .reduce((prev, curr) => prev.concat(curr))
-
-  // Filter out all non-IPv6 addresses
-      .filter((iface) => iface.family === 'IPv6')
-
-  // Map the interface into its address, with interface ID
-      .map((iface) => iface.address + '%' + iface.ifname)
 
-  // Return the first one with a link-local address
-      .find((addr) => addr.startsWith('fe80:'));
+// Try to find an interface with an IPv6 link local address.
+function linklocal() {
+  // Iterate over the OS interfaces.
+  for (const [ifname, entries] of Object.entries(os.networkInterfaces()))
+    // Iterate over the addresses on the interface.
+    for (const { address, family } of entries)
+      // If the address is an IPv6 address and a link local address,
+      if (family === 'IPv6' && address.startsWith('fe80:'))
+        // then return the it, i.e. the first one found.
```

Comment:
While not objectively wrong, I'd rather you drop these comments since they just state in prose what the code on the line below is doing.  Likewise for many other comments in this file.